hal/riscv64/string.c

```diff
@@ -131,7 +101,7 @@ static inline char *hal_strncpy(char *dest, const char *src, size_t n)
 }
 
 
-static inline unsigned int hal_i2s(char *prefix, char *s, unsigned long i, unsigned char b, char zero)
+int hal_i2s(char *prefix, char *s, unsigned int i, unsigned char b, char zero)
```

Comment:
Why `unsigned int` and not `unsigned long`?

Reply:
Changed to `unsigned long` for `riscv` and the other platforms.